File: classes/parte-7-paradigmas/113-oo-basado-en-prototipos-javascript/implementaciones/typescript/main.ts
import { readFileSync } from "node:fs";

const n: number = parseInt(readFileSync(0, "utf8").trim(), 10);
const obj = {
  valor: n,
  doble(): number {
    return this.valor * 2;
  },
};
console.log(`resultado=${obj.doble()}`);

File: classes/parte-7-paradigmas/113-oo-basado-en-prototipos-javascript/implementaciones/rust/main.rs
use std::io::Read;

struct Obj {
    valor: i64,
}

impl Obj {
    fn doble(&self) -> i64 {
        self.valor * 2
    }
}

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let n: i64 = s.trim().parse().unwrap();
    let o = Obj { valor: n };
    println!("resultado={}", o.doble());
}
